plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XLanguageServerImpl.java

```diff
@@ -185,11 +187,7 @@
 
 	private InitializeResult initializeResult;
 
-	private final CompletableFuture<InitializedParams> clientInitialized = new CompletableFuture<>();
-
-	private final CompletableFuture<?> initBuildFinished = new CompletableFuture<>();
-
-	private final CompletableFuture<?> initCleanFinished = new CompletableFuture<>();
+	private final CompletableFuture<InitializedParams> clientInitialized;
```

Comment:
I removed these fields here since we use the same server instance on a clean-rebuild request. These two futures here would already be fulfilled at that point in time. So the methods to join on these were misleading.